zcash_client_sqlite/src/chain.rs

```diff
@@ -1,16 +1,80 @@
 //! Functions for enforcing chain validity and handling chain reorgs.
 use protobuf::Message;
 
-use rusqlite::params;
+use rusqlite::{params};
 
-use zcash_primitives::consensus::BlockHeight;
+use zcash_primitives::{
+    transaction::{components::Amount, TxId},
+    consensus::{self, BlockHeight},
+};
 
-use zcash_client_backend::{data_api::error::Error, proto::compact_formats::CompactBlock};
+use zcash_client_backend::{
+    data_api::error::Error, 
+    proto::compact_formats::CompactBlock,
+    address::RecipientAddress,
+    };
 
 use crate::{error::SqliteClientError, BlockDB};
 
 pub mod init;
 
+
+pub struct UnspentTransactionOutput {
```

Comment:
This type should probably go in zcash-client-backend; we don't actually want any functions depending on any types or calling anything in zcash_client_sqlite directly; the only things that should be used from this crate (not through the DAA) are the WalletDB and BlockDB initialization functions.